### DISCUSSION TITLE
collapsible bootstrap button in quarto

### DISCUSSION THREAD

**Discussion by shamindras** [quarto-dev/quarto-cli#3100] [Q&A]
On my quarto blog, I would like to create a collapsible toggle text button.
In RStudio, I tried to follow [this basic example](https://www.w3schools.com/bootstrap/bootstrap_collapse.asp) and created the following test:

```
<button type="button" class="btn btn-info" data-toggle="collapse" data-target="#demo"><i class="bi-calendar3-week-fill"></i> abstract</button>
<div id="demo" class="collapse">
Lorem ipsum text....
</div>
```

This results in the following being rendered on my blog in RStudio, but unfortunately the clicking the `abstract` button does not toggle expand or collapse to the demo text.

![CleanShot 2022-10-28 at 20 59 36@2x](https://user-images.githubusercontent.com/7627188/198775158-7df4c14d-c061-42c3-ab45-324dec9ce3e0.png)

The actual source for my site [is here](https://github.com/shamindras/ss_personal_quarto_blog/blob/9bf8ae4369064c8f24429ec1b2ea187578b805fc/research.qmd#L74-L91), and this is the resulting [output](https://www.shamindras.com/research.html).

Could anyone please explain how to get this toggle button working?

**Comment by mcanouil** [comment]
You are missing several components.
Since Quarto is using Bootstrap the best place for examples is on the documentation website: <https://getbootstrap.com/docs/5.1/components/collapse/>.

https://user-images.githubusercontent.com/8896044/198824506-bb3acce4-56bc-4e7a-a3f7-63f6b9917d95.mov

<details><summary>Quarto document</summary>

````markdown
---
title: "Untitled"
format: html
---

```{=html}
<button class="btn btn-primary" type="button" data-bs-toggle="collapse" data-bs-target="#collapseExample" aria-expanded="false" aria-controls="collapseExample">
  Button with data-bs-target
</button>
<div class="collapse" id="collapseExample">
  <div class="card card-body">
    Some placeholder content for the collapse component. This panel is hidden by default but revealed when the user activates the relevant trigger.
  </div>
</div>
```
````

</details>

Also, because this is a 100% HTML issue, you'll get better/quicker answer on StackOverflow or other platform.

**Reply by shamindras** [reply]
Thank you @mcanouil for your patient and thoughtful help.

> You are missing several components.
> Since Quarto is using Bootstrap the best place for examples is on the documentation website: [getbootstrap.com/docs/5.1/components/collapse](https://getbootstrap.com/docs/5.1/components/collapse/).

Your `reprex` does work for me now - thanks! I did in fact go to the official page you had mentioned, and should have taken the example from there (with missing components), rather than from the source I noted above.

> Also, because this is a 100% HTML issue, you'll get better/quicker answer on StackOverflow or other platform.

I'll note this for future reference 👍🏾.

**Reply by mcanouil** [reply]
You are welcome, it gave me the opportunity to have a look at Boostrap components (especially, the popover 🤯 <https://getbootstrap.com/docs/5.1/components/popovers/>).

**Reply by shamindras** [reply]
Very cool indeed @mcanouil. I'll def check it out - thanks again.
